### PR TITLE
Introduced script specific errors

### PR DESCRIPTION
**Pull request by natzei**
I've also removed some redundant cases of opcode and refactored a bit CHECKLOCKTIMEVERIFY (added a new method like in BitcoinCore) and CHECKSEQUENCEVERIFY (allowing better specific messages).

I'll wait a feedback before proceeding, i.e. before check that script tests fail with the right error code.

### CODE REVIEW

**Comment by natzei** [comment]
I've completed the implementation :)
All tests passed.

**Comment by natzei** [comment]
Added all the Bitcoin Core MINIMALDATA tests.
This PR solves #1396 and #1397.

File: core/src/main/java/org/bitcoinj/script/ScriptError.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package org.bitcoinj.script;
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/script/ScriptError.java]: For new java files, we need the license.

Comment (natzei) [inline core/src/main/java/org/bitcoinj/script/ScriptError.java]: done

**Comment by natzei** [comment]
Regarding separating the commits, can you suggest how?

File: core/src/main/java/org/bitcoinj/script/ScriptError.java [hunk 2]
```diff
@@ -1,3 +1,19 @@
+/*
+ * Copyright 2013 Google Inc.
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/script/ScriptError.java]: Oh, please don't donate your copyright to Google...

You can simply use "Copyright by the original author or authors." here.

Comment (natzei) [inline core/src/main/java/org/bitcoinj/script/ScriptError.java]: oh... I've copy/pasted without reading.
I've seen that also Script.java contains your copyright. Could I add my name too?

**Comment by schildbach** [comment]
Ok, let's take this example: If I understand your change right, you removed all the case statements that lead into throwing ScriptException("Attempted to use disabled Script Op."). All these cases now run into the default case and will throw ScriptException("Script used a reserved opcode " + opcode).

That change is fine by me, but I think it's unrelated to this PR and should be discussed individually and -- if merged -- end up as a separate commit in the history.

The usage of static imports for VerifyFlags is also an unrelated refactoring, and actually now that I think of it I'd rather not do it because the context of the flags gets lost. (I personally use static imports very rarely, as it tends to obfuscate the code.)

Imho the ultimate goal should be to get the impact of this PR as small as possible, so reviewing is as easy as possible.

File: core/src/main/java/org/bitcoinj/core/Transaction.java [hunk 3]
```diff
@@ -17,32 +17,44 @@
 
 package org.bitcoinj.core;
 
+import static com.google.common.base.Preconditions.checkArgument;
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/Transaction.java]: This "organize imports" seems unnecessary.

**Comment by natzei** [comment]
I agree with you. When refactoring I basically viewed the Bitcoin Core implementation and made them much more similar.
But I know, the impact should be as small as possible. I'll try to fix all the things you mentioned.

File: core/src/main/java/org/bitcoinj/script/ScriptError.java (hunk 2 above)
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/script/ScriptError.java]: If you want you can do that.

**Comment by natzei** [comment]
Done. Please let me know if it's fine now. I'll propose the other changes of Script.java in another PR when this will be merged. Thanks

**Comment by schildbach** [comment]
I removed the two unused constants and merged. You might want to add them in a followup PR. Thanks!

**Pull request closed without merging**